pkg/lang/ir/v1/julia.go

```diff
@@ -18,44 +18,76 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/cockroachdb/errors"
 	"github.com/moby/buildkit/client/llb"
-	"github.com/sirupsen/logrus"
+)
 
-	"github.com/tensorchord/envd/pkg/types"
+const (
+	julia       = "/opt/julia"                                                                           // Location of downloaded Julia binary and other files
+	juliaBin    = "/opt/julia/bin"                                                                       // Location of Julia executable binary file
+	juliaPkg    = "/opt/julia/user_packages"                                                             // Location of additional packages installed via Julia
+	juliaBinary = "https://julialang-s3.julialang.org/bin/linux/x64/1.8/julia-1.8.3-linux-x86_64.tar.gz" // The official link for downloading Julia environment
 )
 
-func (g *generalGraph) installJulia(root llb.State) (llb.State, error) {
-	root = g.updateEnvPath(root, types.DefaultJuliaPath)
-	return root, errors.New("not implemented")
+// getJuliaBinary returns the llb.State only after setting up Julia environment
+// A successful run of getJuliaBinary should set up the Julia environment
+func (g generalGraph) getJuliaBinary(root llb.State) llb.State {
+
+	base := llb.Image(builderImage)
+	builder := base.
+		Run(llb.Shlexf("sh -c \"curl %s -o julia.tar.gz\"", juliaBinary),
```

Comment:
```suggestion
		Run(llb.Shlexf(`sh -c "curl %s -o julia.tar.gz"`, juliaBinary),
```

`julia.tar.gz` could be a const since you need it later.

Reply:
Fixed